### PR TITLE
PE-37978: Add 'amazon' to #repo-filename method

### PR DESCRIPTION
**Pull request by span786**
Added 'amazon' support in #repo_filename method in lib/beaker/host/unix/file.rb.

### CODE REVIEW

**Comment by bastelfreak** [comment]
@span786 can you please rebase so we've a single commit in this PR?

**Comment by span786** [comment]
Hi @bastelfreak, I have squashed the commits into 1.

File: spec/beaker/host/unix/file_spec.rb [hunk 1]
```diff
@@ -133,6 +133,14 @@ def to_s
         expect(filename).to be === correct
       end
 
+      it 'builds the filename correctly for amazon-based platforms' do
+        @platform = 'amazon-2023-x86_64'
```
Comment (TheMeier) [inline spec/beaker/host/unix/file_spec.rb:137]: can this be made date independant?

Comment (ekohl) [reply spec/beaker/host/unix/file_spec.rb:137]: No, the 2023 is the release version. Sort of like how Ubuntu has it.

**Review by ekohl — APPROVED** [review]

**Pull request merged**